deployments/hadoop-yarn/bin/create-users.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/sh
+#
+# <meta:header>
+#   <meta:licence>
+#     Copyright (c) 2022, ROE (http://www.roe.ac.uk/)
+#
+#     This information is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU General Public License as published by
+#     the Free Software Foundation, either version 3 of the License, or
+#     (at your option) any later version.
+#
+#     This information is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#
+#     You should have received a copy of the GNU General Public License
+#     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#   </meta:licence>
+# </meta:header>
+#
+#
+
+    set -eu
+    set -o pipefail
+
+    binfile="$(basename ${0})"
+    binpath="$(dirname $(readlink -f ${0}))"
+    treetop="$(dirname $(dirname ${binpath}))"
+
+    echo ""
+    echo "---- ---- ----"
+    echo "File [${binfile}]"
+    echo "Path [${binpath}]"
+    echo "Tree [${treetop}]"
+
+    cloudbase='arcus'
+    cloudname=${1:?}
+    deployconf="${2:?}"
+
+    inventory="${treetop:?}/hadoop-yarn/ansible/config/${deployconf:?}.yml"
+    deploytype="${3:-prod}"
+
+
+
+    echo "---- ---- ----"
+    echo "Deploy conf [${deployconf}]"
+    echo "Deploy Type [${deploytype}]"
+    echo "---- ---- ----"
+
+
+    # Create the Zeppelin Shiro Database.
+
+    echo ""
+    echo "---- ----"
+    echo "Creating Zeppelin Shiro database"
+
+    pushd "${treetop:?}/hadoop-yarn/ansible"
+
+            ansible-playbook \
+                --inventory "${inventory:?}" \
+                "38-install-user-db.yml"
+    popd
+
+
+
+    # Import users
+
+    pushd "${treetop:?}/hadoop-yarn/ansible"
+
+
+        if [ "${deploytype}" == "test" ]
+        then
+            ansible-playbook \
+                --inventory "${inventory:?}" \
+                --extra-vars "users_import_file=auth-test.sql" \
+                "39-import-users.yml"
+        else
+            ansible-playbook \
+                --inventory "${inventory:?}" \
+                --extra-vars "users_import_file=auth.sql" \
+                "39-import-users.yml"
+        fi
+
+    popd
```

Comment:
Can we change this around a bit, so that a 'prod' deployment will add the original text file based authentication we have now, and a setting deploytype to 'jdbc' installs the JDBCShiro configuration, database and the test user accounts.
That way we can continue to use the automated deployment to deploy the existing system we have now, and we can experiment with the JDBC based deployment by setting deploytype.

Reply:
If we call the parameter `authtype` rather than `deploytype` it makes it clearer what it controls.
```
if [ "${authtype}" == "jdbc"]
then
```

Reply:
I've changed the scripts based on your suggestion, so we can deploy as one of [jdbc, prod, test]
https://github.com/stvoutsin/aglais/commit/aca45e93bdfbe828a2f5a2eb90c59d88ab461b9f

Reply:
Just to check, looking at the if/else in the script is this what it does :

* [authtype=prod]
   * install the original text based authentication
* [authtype=jdbc]
   * install the database authentication
   * install real user accounts
* [authtype=test]
   * install the database authentication
   * install just the test user accounts